blocks/bricks/interfaces.py

```diff
@@ -196,8 +207,10 @@ def _initialize(self):
         # Use self.parameters[] references in case W and b are overridden
         # to return non-shared-variables.
         if self.use_bias:
-            self.biases_init.initialize(self.parameters[1], self.rng)
-        self.weights_init.initialize(self.parameters[0], self.rng)
+            self.initialization_schemes['biases_init'].initialize(
```

Comment:
I think that here you should just initialize each parameter with the scheme corresponding to its most specific role. This assumes that the keys in `initialization_schemes` are the roles. In this case for each parameter you can find the most specific role in `initialization_schemes` and use the scheme given for this roles to initialize the parameter. Note, that with such an approach one does not even have to implement `_initialize` method!